src/main/scala/beam/utils/map/GeoJsonToGpxConvertor.scala

```diff
@@ -2,29 +2,42 @@ package beam.utils.map
 
 import beam.utils.GeoJsonReader
 import com.typesafe.scalalogging.LazyLogging
-import com.vividsolutions.jts.geom.Geometry
+import com.vividsolutions.jts.geom.{Coordinate, Envelope, Geometry}
 import org.matsim.api.core.v01.Coord
 import org.opengis.feature.Feature
 import org.opengis.feature.simple.SimpleFeature
 
 object GeoJsonToGpxConvertor extends LazyLogging {
 
-  def convert(inputPath: String, outputPath: String): Unit = {
-    val start = System.currentTimeMillis()
-    val mapper: (Feature => GpxPoint) = (feature: Feature) => {
+  def renderEnvelope(points: Array[GpxPoint], outputPath: String): Unit = {
+    val envelope = new Envelope(new Coordinate(points.head.wgsCoord.getX, points.head.wgsCoord.getY))
+    points.foreach { p =>
+      envelope.expandToInclude(p.wgsCoord.getX, p.wgsCoord.getY)
+    }
+    new EnvelopeToGpx().render(envelope, None, outputPath)
+  }
+
+  def readGeoJson(inputPath: String): Array[GpxPoint] = {
+    val mapper: Feature => GpxPoint = (feature: Feature) => {
       val movementId = feature.getProperty("MOVEMENT_ID").getValue.toString
       val centroid = feature.asInstanceOf[SimpleFeature].getDefaultGeometry.asInstanceOf[Geometry].getCentroid
       val wgsCoord = new Coord(centroid.getX, centroid.getY)
       GpxPoint(movementId, wgsCoord)
     }
-    val points = GeoJsonReader.read(inputPath, mapper)
-    GpxWriter.write(outputPath, points)
-    val end = System.currentTimeMillis()
-    logger.info(s"Converted '$inputPath' to '$outputPath' in ${end - start} ms")
+    GeoJsonReader.read(inputPath, mapper)
   }
 
   def main(args: Array[String]): Unit = {
-    convert("""C:\temp\movement_data\san_francisco_censustracts.json""", "san_francisco_censustracts.gpx")
-    convert("""C:\temp\movement_data\san_francisco_taz.json""", "san_francisco_taz.gpx")
+    {
+      val censusTractsPoints = readGeoJson("""C:\temp\movement_data\san_francisco_censustracts.json""")
```

Comment:
Should this be part of the final merge? Maybe it could be read from args with a default to something meaningful? Or a helpful hint for those who might try to use it.

Reply:
Let me make it as app args with some comments